add process resource limits to procstat input

phemmer opened the pull request:
This adds resource limits to the procstat input plugin.

closes #2617 

### Required for all PRs:

- [X] Signed [CLA](https://influxdata.com/community/cla/).
- [X] Associated README.md updated.
- [ ] Has appropriate unit tests.

phemmer commented on plugins/inputs/procstat/procstat.go:
```diff
@@ -154,6 +155,48 @@ func (p *Procstat) addMetrics(proc Process, acc telegraf.Accumulator) {
+			fields[prefix+"rlimit_"+name+"_soft"] = rlim.Soft
+			fields[prefix+"rlimit_"+name+"_hard"] = rlim.Hard
+			if name != "file_locks" { // gopsutil doesn't currently track the used file locks count
+				fields[prefix+name] = rlim.Used
```
The convention here is that the field being added is `rlimit_FOO_soft` and `rlimit_FOO_hard` for which there is a field named `FOO` is the current value. Some of these the `FOO` field already existed (e.g. `memory_rss`), and some it is a new field (e.g. `nice_priority`).

danielnelson commented on plugins/inputs/procstat/README.md:
```diff
@@ -53,6 +53,13 @@ Threads related measurement names:
+
+Signals related measurement names:
+- procstat_[prefix_]signals_pending value=0
+
```
Are these 3 actually under `rlimit_*`?

phemmer replied:
yes

danielnelson replied:
Should they be removed here then?

phemmer replied:
No, `procstat_[prefix_]signals_pending` here is the currently used value. The `procstat_[prefix_]rlimit_signals_pending_hard` (and _soft) down in the `rlimit_*` section are the limits.
Or maybe I'm misunderstanding what you mean.

phemmer commented:
Ok, I've updated enough to where it can go through formal review (out of design). Just made the tests pass (circleci failure seems unrelated), and updated docs.

phemmer commented on plugins/inputs/procstat/README.md:
```diff
@@ -64,19 +71,49 @@ I/O related measurement names (*telegraf* needs to run as **root**):
+- procstat_[prefix_]rlimit_signals_pending_hard value=1758
+- procstat_[prefix_]rlimit_signals_pending_soft value=1758
+
+*NOTE: Due to a limitation in an underlying library Telegraf uses, any resource limit > 2147483647 will be misreported as 2147483647.*
```
Oh, and I just remembered to add this little note. I think this is a major issue. But the `gopsutil` owners won't fix the issue because it might break consumers. (which I personally think is a lousy excuse as that's what vendoring is meant to solve)
